keps/2724-topology-aware-scheduling/README.md

```diff
@@ -904,29 +902,180 @@ type TopologyDomainAssignment struct {
+- extracting common prefixes and suffixes of node names.
 
 ```golang
-const (
-  // TopologySchedulingGate is used to delay scheduling of a Pod until the
-  // nodeSelectors corresponding to the assigned topology domain are injected
-  // into the Pod.
-  TopologySchedulingGate = "kueue.x-k8s.io/topology"
+type TopologyAssignment struct {
+  // (same role & comments as in v1beta1)
+  Levels []string
 
-  // WorkloadAnnotation is an annotation set on the Job's PodTemplate to
-  // indicate the name of the admitted Workload corresponding to the Job. The
-  // annotation is set when starting the Job, and removed on stopping the Job.
-  WorkloadAnnotation = "kueue.x-k8s.io/workload"
+  // slices represent topology assignments for subsets of pods of a workload.
+  // The full assignment is obtained as a union of all slices.
+  // +required
+  // +listType=atomic
+  // +kubebuilder:validation:MinItems=1
+  // +kubebuilder:validation:MaxItems=30000
+  Slices []TopologyAssignmentSlice
+}
 
-  // TASLabel is a label set on the Job's PodTemplate to indicate that the
-  // PodSet is admitted using TopologyAwareScheduling, and all Pods created
-  // from the Job's PodTemplate also have the label.
-  TASLabel = "kueue.x-k8s.io/tas"
-)
+// TopologyAssignmentSlice fully specifies the topology assignment for a subset of pods of a workload.
+type TopologyAssignmentSlice struct {
+  // size is the number of domains covered by this slice.
+  // +required
+  // +kubebuilder:validation:Minimum=1
+  Size int32
+
+  // valuesPerLevel has one entry for each of the Levels specified in the TopologyAssignment.
+  // The entry corresponding to a particular level specifies the placement of pods at that level.
+  // +required
+  // +listType=atomic
+  // +kubebuilder:validation:MinItems=1
+  // +kubebuilder:validation:MaxItems=16
+  ValuesPerLevel []TopologyAssignmentSliceLevelValues
+
+  // counts specifies the number of pods allocated per each domain.
+  // May be omitted if all values are identical; if so, UniversalCount is used instead.
+  // If set, its length must be equal to the "size" field.
+  // Exactly one of count, universalCount must be set.
+  // +optional
+  // +listType=atomic
+  // +kubebuilder:validation:MinItems=1
+  // +kubebuilder:validation:MaxItems=100000
+  Counts []int32 `json:"omitempty"`
+
+  // universalCount, if set, specifies the number of pods allocated in every domain in this slice.
+  // Exactly one of count, universalCount must be set.
+  // +optional
+  UniversalCount *int32 `json:"omitempty"`
+}
+
+type TopologyAssignmentSliceLevelValues struct {
+  // commonPrefix specifies a common prefix for all values in this slice assignment.
+  // +optional
+  CommonPrefix *string `json:"omitempty"`
+  // commonSuffix specifies a common suffix for all values in this slice assignment.
+  // +optional
+  CommonSuffix *string `json:"omitempty"`
```

Comment:
```suggestion
  // prefix specifies a common prefix for all values in this slice assignment.
  // +optional
  prefix *string `json:"omitempty"`
  // suffix specifies a common suffix for all values in this slice assignment.
  // +optional
  suffix *string `json:"omitempty"`
```

I would like to recommend avoiding the `common` prefix since it makes API fields more ambiguous, and `common` is ambiguous for targets, like what base group to apply commonization.

Reply:
I don't entirely follow what's unclear - but I'm fine without `Common`. Done.